frontend/public/components/overview.jsx

```diff
@@ -0,0 +1,404 @@
+        leftName = _.get(left, 'metadata.name', '');
+        rightName = _.get(right, 'metadata.name', '');
+        if (descending) {
+          return rightName.localeCompare(leftName);
+        }
+        return leftName.localeCompare(rightName);
+      }
+
+      if (!leftVersion) {
+        return descending ? 1 : -1;
+      }
+
+      if (!rightVersion) {
+        return descending ? -1 : 1;
+      }
+
+      if (descending) {
+        return rightVersion - leftVersion;
+      }
+
+      return leftVersion - rightVersion;
+    };
+
+    return _.toArray(replicationControllers).sort(compareDeployments);
+  }
+
+  filterItems(items) {
+    const {filterValue} = this.state;
+
+    if (!filterValue) {
+      return items;
+    }
+
+    const filterString = filterValue.toLowerCase();
+    return _.filter(items, item => {
+      const name = _.get(item, 'metadata.name', '');
+      return name.toLowerCase().includes(filterString);
+    });
+  }
+
+  groupByLabel(items, label) {
+    return _.chain(items)
+      .groupBy((item) => _.get(item, ['metadata', 'labels', label], 'other'))
+      .map((group, labelValue) => {
+        return {
+          name: labelValue,
+          items: group
+        };
+      })
+      .value();
+  }
+
+  getLabelGroupOptions(items) {
+    const {groupOptions} = this.state;
+    return _.chain(items)
+      .map(item => _.keys(item.metadata.labels))
+      .flatten()
+      .reduce((accumulator, key) => {
+
+        if (_.has(key, accumulator)) {
+          return accumulator;
+        }
+
+        return {
+          ...accumulator,
+          [key]: key
+        };
+      }, groupOptions)
+      .value();
+  }
+
+  buildGraphForReplicators(replicators, kind) {
+    const {pods} = this.props;
+    return _.map(replicators, replicator => {
+      const {uid: replicatorUid} = replicator.metadata;
+      const ownedPods = getOwnedResources(pods.data, replicatorUid);
+      return {
+        ...replicator,
+        kind,
+        pods: ownedPods
+      };
+    });
+  }
+
+  buildGraphForRootResources(rootResources, kind) {
+    const {replicationControllers, replicaSets, routes, services} = this.props;
+
+    return _.map(rootResources, rootResource => {
+      const {uid: rootResourceUid} = rootResource.metadata;
+      // Determine the replication controllers associated with this DC
+      const ownedReplicationControllers = this.buildGraphForReplicators(getOwnedResources(replicationControllers.data, rootResourceUid), 'ReplicationController');
+      const ownedReplicaSets = this.buildGraphForReplicators(getOwnedResources(replicaSets.data, rootResourceUid), 'ReplicaSet');
+      const orderedReplicationControllers = this.sortByDeploymentVersion(ownedReplicationControllers, true);
+      const currentController = _.head(orderedReplicationControllers) || _.head(ownedReplicaSets);
+      const prevController = orderedReplicationControllers[1];
+
+      // Get the routes for this deployment config
```

Comment:
I would take this out for now if we're not showing routes or services yet. (Save the code in a branch :)